File: Examples/DotNET/CSharp/Files/MoveFile.cs
using System;
using Com.Aspose.Storage.Api;
using Com.Aspose.Storage.Model;

namespace Files
{
    class MoveFile
    {
        public static void Run()
        {
            // ExStart:1           
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            string Path = "testfile.txt";
            string dest = "new-testfile.txt";
            string versionId = null;
            string storage = null;
            string destStorage = null;        
            byte[] file = System.IO.File.ReadAllBytes(Common.GetDataDir() + Path);
            try
            {
                // Invoke Aspose.Storage Cloud SDK API to upload and move file
                storageApi.PutCreate(Path, versionId, storage, file);
                MoveFileResponse apiResponse  = storageApi.PostMoveFile(Path, dest, versionId, storage, destStorage);
                if (apiResponse != null && apiResponse.Status.Equals("OK"))
                {
                    Console.WriteLine("File moved successfully.");
                    Console.ReadKey();
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);
            }
            // ExEnd:1
        }
    }
}

File: Examples/Ruby/Files/move_file.rb
#ExStart:
require 'aspose_storage_cloud'

class Files
    
  include AsposeStorageCloud

  def initialize
    # Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("App Key", "App SID")
    @storage_api = StorageApi.new
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new 
    response = @storage_api.put_create(file_name, File.open("../../../data/" << file_name,"r") { |io| io.read } )
  end
  
  def create_folder(folder_name)
    @storage_api = StorageApi.new
    response = @storage_api.put_create_folder(folder_name)
  end

  def move_file
    # Upload file to Aspose Cloud Storage
    file_name = "testfile.txt"
    upload_file(file_name)

    folder_name = "Files"
    create_folder(folder_name)

    dest_folder = "#{folder_name}/#{file_name}"

    response = @storage_api.post_move_file(file_name, dest_folder)
  end

end

files = Files.new()
puts files.move_file
#ExEnd: